bytecomp/bytegen.ml

```diff
@@ -465,12 +465,17 @@ let comp_primitive p args =
   | Pisout -> Kisout
   | Pbintofint bi -> comp_bint_primitive bi "of_int" args
   | Pintofbint bi -> comp_bint_primitive bi "to_int" args
-  | Pcvtbint(Pint32, Pnativeint) -> Kccall("caml_nativeint_of_int32", 1)
-  | Pcvtbint(Pnativeint, Pint32) -> Kccall("caml_nativeint_to_int32", 1)
-  | Pcvtbint(Pint32, Pint64) -> Kccall("caml_int64_of_int32", 1)
-  | Pcvtbint(Pint64, Pint32) -> Kccall("caml_int64_to_int32", 1)
-  | Pcvtbint(Pnativeint, Pint64) -> Kccall("caml_int64_of_nativeint", 1)
-  | Pcvtbint(Pint64, Pnativeint) -> Kccall("caml_int64_to_nativeint", 1)
+  | Pcvtbint(src, dst) ->
+      begin match (src, dst) with
+      | (Pint32, Pnativeint) -> Kccall("caml_nativeint_of_int32", 1)
+      | (Pnativeint, Pint32) -> Kccall("caml_nativeint_to_int32", 1)
+      | (Pint32, Pint64) -> Kccall("caml_int64_of_int32", 1)
+      | (Pint64, Pint32) -> Kccall("caml_int64_to_int32", 1)
+      | (Pnativeint, Pint64) -> Kccall("caml_int64_of_nativeint", 1)
+      | (Pint64, Pnativeint) -> Kccall("caml_int64_to_nativeint", 1)
+      | ((Pint32 | Pint64 | Pnativeint), _) ->
+          fatal_error "Bytegen.comp_primitive: invalid Pcvtbint cast"
```

Comment:
A simpler (smaller diff, less nesting) change would be to add a toplevel case
```
| Pcvtbint(Pint32 | Pint64 | Pnativeint, _) ->
    fatal_error "Bytegen;comp_primitive: invalid Pcvtbint cast"
```

Reply:
I have a preference for the nested form, which clearly handles all `Pcvtbint` cases together, instead of having to inspect the outer pattern to tell that all cases are handled. Would you like it better if it was moved into an auxiliary function? (Naming suggestions?)

Reply:
Nah, it was just a possibility, I'm also happy with the code as it is.